File: eRecipes/UI/erecipes_desktop/lib/models/katalog.dart
import 'package:erecipes_desktop/models/katalog_recept.dart';
import 'package:json_annotation/json_annotation.dart';

part 'katalog.g.dart';

@JsonSerializable()
class Katalog {
  int? katalogId;
  String? naziv;
  String? opis;
  DateTime? datumKreiranja;
  List<KatalogRecept>? katalogRecepts;

  Katalog();

  factory Katalog.FromJson(Map<String, dynamic> json) =>
      _$KatalogFromJson(json);

  Map<String, dynamic> toJson() => _$KatalogToJson(this);
}

File: eRecipes/eRecipes.Model/Katalog.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eRecipes.Model
{
    public class Katalog
    {
        public int KatalogId { get; set; }
        public string Naziv { get; set; } = null!;
        public DateTime DatumKreiranja { get; set; }
        public string? Opis { get; set; }
        public ICollection<KatalogRecept> KatalogRecepts { get; set; } = new List<KatalogRecept>();
    }
}
